interfaces.go

```diff
@@ -120,6 +120,9 @@ type CallMsg struct {
 	Value    *big.Int        // amount of wei sent along with the call
 	Data     []byte          // input data, usually an ABI-encoded contract method invocation
 
+	FeeCap *big.Int // EIP-1559 fee cap per gas.
+	Tip    *big.Int // EIP-1559 tip per gas.
```

Comment:
Since these fields are per gas, wouldn't it make sense to name them GasFeeCap and GasTip? The naming is ever slightly more unwieldy, but I think it makes it clearer that it's per gas unit, not a total. CC @holiman?

Also, if the ordering of the fields here is not relevant, I'd move it up below GasPrice, since contextually that seems saner and also follows the transaction field consensus layout.

Reply:
Also note, the EIP-1559 spec refers to `FeeCap` as `MaxFeePerGas` and `Tip` as `PriorityFeePerGas`. I find the shorter versions easier to reason about, but it may be too much of a headache in the long run.

Reply:
Further discussed here by geth team: https://gist.github.com/karalabe/1565e0bc1be6895ad85e2a0116367ba6